ls1intum/Artemis
# `Development`: Differentiate between buttons and links

### Checklist
#### General



#### Client
- [x] I added screenshots of the UI 


### Motivation and Context
closes #7564 

### Description
simplify tutorial groups using relative links and adding some logic to differentiate when lecture units updated within guided mode (we need buttons then) an when in an extra page (-> links).

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students
- 1 Lecture
- 1 Tutorial Group

1. Log in to Artemis as Instructor
2. Navigate to Lecture Management.
3. Verify that it is possible to edit/create lecture units both in guided mode for lecture editing & in lecture-unit page
4. Verify that the buttons for creating/editing on the lecture-unit page are html-a tags (you should see a path preview in your browser when hovering on them -> at least in chrome)
5. Navigate to tutorial Group Management Verify that you can navigate through the tutorial groups and all links on the Detail Page work (Course Link, tutorial Group Link, should also be HTML a tags now)
6. Log in as student
7. Navigate through Tutorial Groups
8. Verify that all links work (Course Link, tutorial group link should be HTML a tags now)


### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [tutorial-group-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5729/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/tutorial-groups/shared/tutorial-group-detail/tutorial-group-detail.component.ts.html) | 95.65% | ✅  |
| [tutorial-group-row.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5729/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/tutorial-groups/shared/tutorial-groups-table/tutorial-group-row/tutorial-group-row.component.ts.html) | 100% | ✅ |
| [tutorial-groups-table.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5729/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/tutorial-groups/shared/tutorial-groups-table/tutorial-groups-table.component.ts.html) | 90.24% | ✅  |
| [tutorial-group-management-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5729/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/tutorial-groups/tutorial-groups-management/tutorial-groups/detail/tutorial-group-management-detail.component.ts.html) | 96.77% | ✅  |
| [tutorial-groups-management.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5729/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/tutorial-groups/tutorial-groups-management/tutorial-groups/tutorial-groups-management/tutorial-groups-management.component.ts.html) | 91.83% | ✅  |
| [lecture-unit-management.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5729/latest/artifact/shared/Coverage-Report-Client-Tests/app/lecture/lecture-unit/lecture-unit-management/lecture-unit-management.component.ts.html) | 89.24% | ✅  |
| [unit-creation-card.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5729/latest/artifact/shared/Coverage-Report-Client-Tests/app/lecture/lecture-unit/lecture-unit-management/unit-creation-card/unit-creation-card.component.ts.html) | 100% | ✅ |
| [course-tutorial-group-card.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5729/latest/artifact/shared/Coverage-Report-Client-Tests/app/overview/course-tutorial-groups/course-tutorial-group-card/course-tutorial-group-card.component.ts.html) | 100% | ✅ |
| [course-tutorial-groups-overview.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5729/latest/artifact/shared/Coverage-Report-Client-Tests/app/overview/course-tutorial-groups/course-tutorial-groups-overview/course-tutorial-groups-overview.component.ts.html) | 100% | ✅ |
| [course-tutorial-group-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5729/latest/artifact/shared/Coverage-Report-Client-Tests/app/overview/tutorial-group-details/course-tutorial-group-detail/course-tutorial-group-detail.component.ts.html) | 95.65% | ✅  |



### Screenshots
![image](https://github.com/ls1intum/Artemis/assets/78978542/bceeab11-0b55-4155-8ff2-dc1491c953e3)
![image](https://github.com/ls1intum/Artemis/assets/78978542/97bd53a3-0ca3-4b75-8748-52378d031dd2)
